Find a functionally equivalent code:android.widget.CalendarView.setMaxDate:COMMENT
Method Modifier: public      
Comment:/**
 * Sets the maximal date supported by this {@link CalendarView} in milliseconds
 * since January 1, 1970 00:00:00 in {@link TimeZone#getDefault()} time
 * zone.
 *
 * @param maxDate The maximal supported date.
 *
 * @attr ref android.R.styleable#CalendarView_maxDate
 */

@@ -1,19 +1,3 @@
 {
-    mTempDate.setTimeInMillis(maxDate);
-    if (isSameDate(mTempDate, mMaxDate)) {
-        return;
-    }
-    mMaxDate.setTimeInMillis(maxDate);
-    // reinitialize the adapter since its range depends on max date
-    mAdapter.init();
-    Calendar date = mAdapter.mSelectedDate;
-    if (date.after(mMaxDate)) {
-        setDate(mMaxDate.getTimeInMillis());
-    } else {
-        // we go to the current date to force the ListView to query its
-        // adapter for the shown views since we have changed the adapter
-        // range and the base from which the later calculates item indices
-        // note that calling setDate will not work since the date is the same
-        goTo(date, false, true, false);
-    }
+    mDelegate.setMaxDate(maxDate);
 }

